libraries/AP_BLHeli/AP_BLHeli.cpp

```diff
@@ -590,6 +593,9 @@ bool AP_BLHeli::BL_SendBuf(const uint8_t *buf, uint16_t len)
         blheli.ack = ACK_D_GENERAL_ERROR;
         return false;
     }
+    // 19200 baud is 52us per bit - wait for half a bit between sending and receiving to avoid reading
+    // the end of the last sent bit by accident
+    hal.scheduler->delay_microseconds(26);
```

Comment:
can't we make a more generic function that flush and wait half a bit automatically based on the baurate into the RCOutput or serial class ? I am sure this can be handly for some other use or is already in use elsewhere

Reply:
Not sure it buys us much other than complexity. I came up with this value by looking Saleae traces, everywhere else we rely on ChibiOS to provide half-duplex support so not sure there are other use cases